subiquitycore/tests/test_async_helpers.py

```diff
@@ -45,3 +45,24 @@ async def fn():
         sit.task.cancel()
         self.assertEqual(expected_call_count, mock_fn.call_count)
         self.assertEqual(cancel_restart, restarted)
+
+
+# previously, wait() may or may not have been safe to call, depending
+# on if the task had actually been created yet.
+class TestSITWait(unittest.IsolatedAsyncioTestCase):
+    async def test_wait_started(self):
+        async def fn():
+            pass
+        sit = SingleInstanceTask(fn)
+        await sit.start()
+        await asyncio.wait_for(sit.wait(), timeout=1.0)
+        self.assertTrue(sit.done())
+
+    async def test_wait_not_started(self):
+        async def fn():
+            await asyncio.sleep(5)
+        sit = SingleInstanceTask(fn)
+        self.assertFalse(sit.done())
+        with self.assertRaises(TimeoutError):
```

Comment:
Until python 3.11, `asyncio.TimeoutError` and `TimeoutError` were two distinct types.

Using `asyncio.TimeoutError` should be backward compatible I think.

Reply:
Thank you!